app/PaymentDrivers/BlockonomicsPaymentDriver.php

```diff
@@ -107,17 +107,16 @@ public function processWebhookRequest(PaymentWebhookRequest $request)
         $addr = $request->addr;
 
         $payment = Payment::query()
-                            ->where('company_id', $company->id)
-                            ->where('transaction_reference', $txid)
-                            ->firstOrFail();
+            ->where('company_id', $company->id)
+            ->where('custom_value1', $txid)
+            ->where('custom_value2', $addr)
+            ->firstOrFail();
```

Comment:
Is this necessary to combine custom fields into the query like this? if a user is using payment custom fields for another purpose, this would block their usage across the application.

Reply:
I've updated the logic, to only do this for test payments. Regular payments will continue to use the previous method. Hopefully that's a good solution. Please let me know if you have any other suggestions